src/main/java/com/google/sps/servlets/UserDataServlet.java

```diff
@@ -27,44 +27,33 @@
 @WebServlet("/user")
 public class UserDataServlet extends HttpServlet {
 
-  /** Responsible for creating new User. */
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  /** Responsible for add a new User into Datastore */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     // Get the input from the form.
-    String name = getParameter(request, "name", "");
-    String email = getParameter(request, "email", "");
-    String university = getParameter(request, "university", "");
+    String name = request.getParameter("name");
+    String email = request.getParameter("email");
+    String university = request.getParameter("university");
     String subscription = request.getParameter("email-notif");
-    boolean subscribe;
+    boolean subscribe = true;
 
-    if (subscription.equals("subscribe")) {
-      subscribe = true;
-    } else if (subscription.equals("unsubscribe")) {
+    // Check if user wants to unsubscribe.
+    if (subscription.equals("unsubscribe")) {
       subscribe = false;
     }
 
     // Create an Entity type User.
-    Entity userEntity = new Entity("User");
+    Entity userEntity = new Entity("User", email);
     userEntity.setProperty("name", name);
-    userEntity.setProperty("email", email);
     userEntity.setProperty("university", university);
+    userEntity.setProperty("subscribe", subscribe);
 
-    // Store the User in Datastore and refresh page.
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    // Store the User in Datastore and Datastore automatically checks for duplicates
+    // and will update the user information accordingly to new information based on emails
```

Comment:
E.g. a comment like this could just be 'This is fine since Datastores updates the entity if it existed before'

Reply:
Changed!